src/overlay_12_0224E4FC.c

```diff
@@ -0,0 +1,1224 @@
+            if (!flag) {
+                ctx->battleMons[battlerId1].unk88.quickClawFlag = TRUE;
+            }
+        }
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_PINCH_PRIORITY) {
+        if (GetBattlerAbility(ctx, battlerId1) == ABILITY_GLUTTONY) {
+            extra1 /= 2;
+        }
+        if (ctx->battleMons[battlerId1].hp <= ctx->battleMons[battlerId1].maxHp/extra1) {
+            boostedPriority1 = 1;
+            if (!flag) {
+                ctx->battleMons[battlerId1].unk88.custapBerryFlag = TRUE;
+            }
+        }
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_SPEED_DOWN) {
+        loweredPriority1 = 1;
+    }
+    
+    for (i = 0; i < NELEMS(sSpeedHalvingItemEffects); i++) {
+        if (GetItemHoldEffect(ctx, ctx->battleMons[battlerId2].item, 1) == sSpeedHalvingItemEffects[i]) {
+            speed2 /= 2;
+            break;
+        }
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_CHOICE_SPEED) {
+        speed2 = speed2 * 15/10;
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_DITTO_SPEED_UP && ctx->battleMons[battlerId2].species == SPECIES_DITTO) {
+        speed2 *= 2;
+    }
+    
+    if (ability2 == ABILITY_QUICK_FEET && ctx->battleMons[battlerId2].status & 0xFF) {
+        speed2 = speed2 * 15/10;
+    } else if (ctx->battleMons[battlerId2].status & STATUS_PARALYSIS) {
+        speed2 /= 4;
+    }
+    
+    if (ability2 == ABILITY_SLOW_START && ctx->totalTurns - ctx->battleMons[battlerId2].unk88.slowStartTurns < 5) {
+        speed2 /= 2;
+    }
+    
+    if (ability2 == ABILITY_UNBURDEN && ctx->battleMons[battlerId2].unk88.knockOffFlag && ctx->battleMons[battlerId2].item == FALSE) {
+        speed2 *= 2;
+    }
+
+    if (ctx->fieldSideConditionFlags[BattleSys_GetFieldSide(bsys, battlerId2)] & SIDE_CONDITION_TAILWIND) {
+        speed2 *= 2;
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_SOMETIMES_PRIORITY) {
+        if ((ctx->unk_310C[battlerId2])%(100/extra2) == 0) {
+            boostedPriority2 = 1;
+            
+            if (!flag) {
+                ctx->battleMons[battlerId2].unk88.quickClawFlag = TRUE;
+            }
+        }
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_PINCH_PRIORITY) {
+        if (GetBattlerAbility(ctx, battlerId2) == ABILITY_GLUTTONY) {
+            extra2 /= 2;
+        }
+        if (ctx->battleMons[battlerId2].hp <= ctx->battleMons[battlerId2].maxHp/extra2) {
+            boostedPriority2 = 1;
+            if (!flag) {
+                ctx->battleMons[battlerId2].unk88.custapBerryFlag = TRUE;
+            }
+        }
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_SPEED_DOWN) {
+        loweredPriority2 = 1;
+    }
+    
+    ctx->unk_21F0[battlerId1] = speed1;
+    ctx->unk_21F0[battlerId2] = speed2;
+    
+    if (!flag) {
+        action1 = ctx->unk_21A8[battlerId1][3];
+        action2 = ctx->unk_21A8[battlerId2][3];
+        movePos1 = ctx->movePos[battlerId1];
+        movePos2 = ctx->movePos[battlerId2];
+        if (action1 == 1) { //fight button
```

Comment:
constants?

Reply:
maybe? I'd have to see what the other consts would be when I find out more about unk_21A8